server/src/test/java/org/elasticsearch/index/mapper/KeywordFieldMapperTests.java

```diff
@@ -797,4 +799,15 @@ public void testDocValuesLoadedFromStoredSynthetic() throws IOException {
         );
         assertScriptDocValues(mapper, "foo", equalTo(List.of("foo")));
     }
+
+    public void testExactQueryWithNormalizer() throws IOException {
+        // When a keyword is configured with a normalizer, exact queries fall back to
+        // checking the source
+        MapperService mapper = createMapperService("""
+            { "_doc" : { "properties" : { "field" : { "type" : "keyword", "normalizer" : "lowercase" } } } }
+            """);
+        SearchExecutionContext sec = createSearchExecutionContext(mapper);
+        Query q = mapper.fieldType("field").exactQuery("value", sec);
+        assertThat(q, instanceOf(TextFieldExactQuery.class));
+    }
```

Comment:
Maybe also test the case when no normalizer is configured, to make sure we use a `TermQuery` rather than a query that may read the `_source`?